### PR TITLE
Simplify .gitignore for C tests

### PR DESCRIPTION
**Pull request by DaoWen**

### CODE REVIEW

**Comment by DaoWen** [comment]
We weren't ignoring the `boot0` or `deadlock0` test binaries. The more general wildcards make it so we won't need to modify the .gitignore when adding new tests unless the prefix is totally new.

**Comment by budimlic** [comment]
What happened to *.dSYM?

**Comment by DaoWen** [comment]
> What happened to *.dSYM?

It's no longer needed since `async0.dSYM` is covered by the `async*` rule, etc.

**Comment by budimlic** [comment]
Gotcha. I would only add another rule for not excluding .h files (which we currently don't have in the test directory, but might in the future)

File: test/c/.gitignore [hunk 1]
```diff
@@ -1,20 +1,8 @@
...
-forasync2DRec
-forasync3DCh
-forasync3DRec
+async*
+boot*
+deadlock*
+future*
+finish*
+forasync*
+!*.[ch]
```
Comment (DaoWen) [inline test/c/.gitignore:8]: Now we exclude `*.c` and `*.h` files from being ignored.

**Comment by agrippa** [comment]
Looks good to me. @budimlic can merge once he signs off.

**Pull request merged**